In [ ]:
from sklearn.datasets import load_diabetes
import pandas as pd
import numpy as np
diabetes = load_diabetes()
df = pd.DataFrame(diabetes.data, columns=diabetes.feature_names, dtype=np.float32)
df.tail(3)
y = diabetes.target.astype(np.float32)

age       sex       bmi        bp        s1        s2        s3  \
439  0.041708  0.050680 -0.015906  0.017282 -0.037344 -0.013840 -0.024993   
440 -0.045472 -0.044642  0.039062  0.001215  0.016318  0.015283 -0.028674   
441 -0.045472 -0.044642 -0.073030 -0.081414  0.083740  0.027809  0.173816   

           s4        s5        s6  
439 -0.011080 -0.046879  0.015491  
440  0.026560  0.044528 -0.025930  
441 -0.039493 -0.004220  0.003064

위의 데이터는 어떠한 특징들을 가지고 있는 환자들의 당뇨병 진행정도를 파악하는 데이터입니다.

총 데이터는 441개가 있고 특징은 10개가 있으며 y데이터에는 당뇨병 진행정도를 나타내는 값이 들어있습니다.

1. 학습을 위해 다섯개의 특징을 가진 특징 집합 (특징집합 A,B) 두개를 선택합니다. 집합 A와 집합 B는 많아야 두개의 특징을 공유할 수 있습니다.
2. 이 데이터에 대해서 regression 문제를 푸는 신경망을 정의하고 집합 A와 집합 B 그리고 tensorflow를 이용해서 학습을 진행합니다.
3. 동일한 테스트 데이터 100개를 이용해서 집합 A와 집합 B를 이용해서 학습한 신경망을 테스트하여 두 집합 중 어느것이 regresison을 위해 좋은 데이터인지 판단해 보시오. 판정을 위해 Loss 함수 값을 이용하시오.

참고. linear regression 문제는 예측값과 정답값이 같으면 오히려 과적합 문제일 수 있습니다. 따라서 학습한 모델의 예측값과 정답값의 차이가 있더라도 엄청난 차이가 아닌 이상 정상입니다.